charts/unleash/templates/deployment.yaml

```diff
@@ -4,40 +4,40 @@ kind: Deployment
 metadata:
   name: {{ include "unleash.fullname" . }}
   labels:
-{{ include "unleash.labels" . | indent 4 }}
+    {{- include "unleash.labels" . | nindent 4 }}
 spec:
   replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "unleash.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- include "unleash.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       annotations:
         {{-  range $key, $value := .Values.podAnnotations }}
         {{ $key }}: {{ $value | quote }}
         {{- end }}
       labels:
-        app.kubernetes.io/name: {{ include "unleash.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
+        {{- include "unleash.selectorLabels" . | nindent 8 }}
         {{-  range $key, $value := .Values.podLabels }}
         {{ $key }}: {{ $value | quote }}
         {{- end }}
     spec:
-    {{- with .Values.imagePullSecrets }}
+      {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
-    {{- end }}
+      {{- end }}
       containers:
         - name: {{ .Chart.Name }}
           env:
+            {{- if or (.Values.dbConfig.useExistingSecret.name) (.Values.postgresql.enabled) }}
             - name: POSTGRESQL_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Release.Name }}-postgresql
-                  key: postgresql-password
+                  name: {{ if .Values.dbConfig.useExistingSecret.name }}{{ .Values.dbConfig.useExistingSecret.name }}{{ else }}{{ .Values.postgresql.fullnameOverride }}{{ end }}
+                  key: {{ if .Values.dbConfig.useExistingSecret.key }}{{ .Values.dbConfig.useExistingSecret.key }}{{ else }}postgresql-password{{ end }}
+            {{- end }}                  
             - name: DATABASE_URL
-              value: postgres://postgres:$(POSTGRESQL_PASSWORD)@{{ .Release.Name }}-postgresql/unleash
+              value: postgres://{{ .Values.dbConfig.user }}:{{ if or (.Values.dbConfig.useExistingSecret.name) (.Values.postgresql.enabled) }}$(POSTGRESQL_PASSWORD){{ else }}{{ .Values.dbConfig.pass }}{{ end }}@{{ if .Values.postgresql.enabled }}{{ .Values.postgresql.fullnameOverride }}{{ else }}{{ .Values.dbConfig.host }}{{ end }}:{{ .Values.dbConfig.port }}/{{ .Values.dbConfig.database }}
```

Comment:
This line is pretty long. Perhaps we should switch to using DB_USER, DB_PASS, DB_HOSTNAME, DB_NAME and then build DATABASE_URL (if that's what Unleash is expecting) in the template.yaml?

Reply:
Not blocking the PR for this - it's just a comment.

Reply:
I'll try to change it tomorrow :)